cert/tls.go

```diff
@@ -58,3 +107,41 @@ func TLSConfFromCert(certData tls.Certificate) *tls.Config {
 		MinVersion:   tls.VersionTLS12,
 	}
 }
+
+// NewTLSReloader is used to create a new TLS Reloader that will be used
+// to update the TLS certificate without restarting the server.
+func NewTLSReloader(certBytes, keyBytes []byte) (*TlsReloader, error) {
+	result := &TlsReloader{}
+	cert, _, err := LoadCertFromBytes(certBytes, keyBytes)
+	if err != nil {
+		return nil, err
+	}
+	result.cert = &cert
+	return result, nil
+}
+
+// AttemptReload will make an attempt to update the TLS certificate
+// and key used by the server.
+func (tlsr *TlsReloader) AttemptReload(certBytes, keyBytes []byte) error {
+	newCert, _, err := LoadCertFromBytes(certBytes, keyBytes)
+	if err != nil {
+		return err
+	}
+	tlsr.certMu.Lock()
+	defer tlsr.certMu.Unlock()
```

Comment:
The locks should be visually grouped (empty lines above and below) to stand out clearly.
And in this case it could just be:
```
t.certMu.Lock()
t.cert = &newCert
t.certMu.Unlock()
```

But I guess that comes down to personal tatest.
```